next/components/forms/widgets/UploadWidgetRJSF.tsx

```diff
@@ -0,0 +1,108 @@
+import { UploadMinioFile } from '@backend/dtos/minio/upload-minio-file.dto'
+import { StrictRJSFSchema, WidgetProps } from '@rjsf/utils'
+import cx from 'classnames'
+import { useEffect, useState } from 'react'
+
+import Upload from '../Upload/Upload'
+import UploadRJSFOptions from '../Upload/UploadRJSFOptions'
+
+interface UploadWidgetRJSFProps extends WidgetProps{
+  options: UploadRJSFOptions
+  schema: StrictRJSFSchema
+  value: string|string[]
+  disabled?: boolean
+  multiple?: boolean
+  onChange: (value: string|string[]) => void;
+}
+
+const UploadWidgetRJSF = (props: UploadWidgetRJSFProps) => {
+  const {
+    options,
+    schema,
+    value,
+    disabled,
+    onChange
+  } = props
+
+  const {
+    size,
+    accept,
+    type = "button",
+    className
+  } = options
+
+  const supportedFormats = accept?.split(",")
+  const multiple = schema.type === "array"
+  const widgetClassName = cx("px-2 py-4", className)
+
+  const fileNameToMinioFile = (fileName: string): UploadMinioFile => {
+    const fileNameArray = fileName.split("_")
+    fileNameArray.splice(0,2)
+    const originalName = fileNameArray.join("_")
+    console.log(originalName)
+    return {
+      file: new File([], fileName),
+      isUploading: false,
+      originalName
+    }
+  }
+
+  const transformInitialValue = () => {
+    const files: UploadMinioFile[] = []
+
+    if (!value || value.length === 0) {
+      return files
+    } else if (schema.type === "string" && typeof value === 'string') {
+      const minioFile = fileNameToMinioFile(value)
+      files.push(minioFile)
+    } else if (schema.type === "array" && typeof value !== 'string') {
+      value.forEach(file => {
+        const minioFile = fileNameToMinioFile(file)
+        files.push(minioFile)
+      })
+    }
+
+    return files
+  }
```

Comment:
no need for the function, just calculate files on each rerender. You can also simplify it quite a bit, something like (did not run this, may need recheck):

```suggestion
    const valueArray = schema.type === "array"  && Array.isArray(value) ? value : value ? [value] : [] 
    const files: UploadMinioFile[] = valueArray.map(fileNameToMinioFile)
```

afterwards use files directly instead of the function you've used previously